### ISSUE TITLE
Not working after updates

### ISSUE THREAD

**Issue by ghost** [GeyserMC/MCProtocolLib#101]
Today i found that after several new mcprotocollib snapshots my integration tests started failing. Looks like client not connection at all. Please fix this. And please, create stable releases that will not ruin other users code after each update. thx.

``` java
    @Test
    public void testClient() {
        Server server = new Server();
        InetSocketAddress address = new InetSocketAddress("localhost", 25560);
        assertTrue(server.start(address));
        assertTrue(server.isOnline());

        Session session = new Client("localhost", 25560, new MinecraftProtocol("ensirius"), new TcpSessionFactory()).getSession();
        session.connect(true);

        assertTrue(session.isConnected()); // <--- FALSE
    }
```

```
java.lang.AssertionError
```

**Comment by Steveice10** [comment]
Fixed: https://github.com/Steveice10/PacketLib/commit/4c06294c883e78581a686f3aafd287aa912ce9ae

**Comment by ghost** [comment]
I am sorry, but fresh snapshot not helped for me.

``` java
    public void connect() {
        MinecraftProtocol protocol = null;

        if(password != null) {
            try {
                protocol = new MinecraftProtocol(name, password, false);
            } catch (RequestException e) {
                Throwables.propagate(e);
            }
        } else {
            protocol = new MinecraftProtocol(name);
        }

        session = new Client("localhost", TestServer.getPort(), protocol, new TcpSessionFactory()).getSession();
        session.addListener(this); // <-- Already has Disconnected event.
        session.addListener(new SessionAdapter() {
            @Override
            public void disconnected(DisconnectedEvent event) {
                System.out.println("Disconnected " + event.getReason());
            }
        });
        session.connect(true);

        assertTrue("Failed connect to server", session.isConnected());
    }
```

```
java.lang.AssertionError: Failed connect to server
```

**Comment by Steveice10** [comment]
Could you add a listener to the session before connecting to see if DisconnectedEvent is being called with a reason and/or error?

**Comment by ghost** [comment]
Modified my snippet. No disconnectedevent called.

**Comment by Steveice10** [comment]
And you are sure you're using the latest code? The included example uses isConnected in a while loop to wait for the status query to complete, and it works fine.

**Comment by ghost** [comment]
Yes, i am sure. I forgot to tell, that status query test working for me.

**Comment by ghost** [comment]
Looks like problem may be on my side. Because #102 passes all tests. I found that client not showing anything if server in online mode and client trying join as offline.

UPD:
Problem may be because something happens during connect() before connect, you drop exception, that call disconnect event, but to call disconnect event channel must be init. (disconnected = true).

UPD2:
Problem fixed by upgrading my netty from 5.0.0-Alpha1 to 5.0.0-Alpha2

UPD3:
Problem in this throwable https://github.com/Steveice10/PacketLib/blob/4c06294c883e78581a686f3aafd287aa912ce9ae/src/main/java/org/spacehq/packetlib/tcp/TcpClientSession.java#L124

``` java
java.lang.NoSuchMethodError: io.netty.bootstrap.Bootstrap.channel(Ljava/lang/Class;)Lio/netty/bootstrap/AbstractBootstrap;
    at org.spacehq.packetlib.tcp.TcpClientSession.connect(TcpClientSession.java:52)
    at ru.ensemplix.server.TestBot.connect(TestBot.java:71)
```

**Comment by Steveice10** [comment]
Not sure if that's the error you were getting or if you're still getting it, but if you're still getting it make sure your environment is set up properly and that Netty is the right version.
